src/pages/loginform.js

```diff
@@ -0,0 +1,66 @@
+
+import "../styles/input.css";
+import "../styles/button.css";
+
+import {useNavigate} from "react-router-dom";
+import {useState} from "react";
+import { useAuth,userCredentials } from "../context/authcontext";
+
+function Loginform(){
+
+   
+
+    function credentialVerification(username,password){
+
+        if(username===userCredentials.usernname && password===userCredentials.password){
+            return true;
+        }
+        return false;
+    }
+
+    const [userCred,setUserCreds]=useState({username:"",password:""});
+    const navigate=useNavigate();
+
+
+    const {state,dispatch}=useAuth();
+
```

Comment:
Keep this at start of function Loginform and after this keep helper functions which are used i.e credentialVerification()